### PR TITLE
Findobj

### PR DESCRIPTION
**Pull request by DLWoodruff**
Dropping dependence on PySP  by adding my version of an objective function finder. This fixes issue #9

### CODE REVIEW

File: mpisppy/utils/sputils.py [hunk 1]
```diff
@@ -744,6 +744,20 @@ def reenable_tictoc_output():
     # Primarily to re-enable after a disable
     tt_timer._ostream.close()
     tt_timer._ostream = sys.stdout
+
+    
+def find_active_objective(pyomomodel):
```
Comment (bknueven) [inline mpisppy/utils/sputils.py:749]: I think we could simplify the implementation of this function:

retval = None
for obj in (...):
    if retval is None:
        retval = obj
    else:
        raise RuntimeError(...)
return retval

**Review by bknueven — APPROVED** [review]
Looks good

File: mpisppy/utils/sputils.py (hunk 1 above)
Comment (bknueven) [reply mpisppy/utils/sputils.py:749]: Maybe we should also be raising an exception if there's *no* objective..

Comment (jsiirola) [reply mpisppy/utils/sputils.py:749]: This seems excessively complicated.  What do you think of the following:
```python
 def find_active_objective(pyomomodel):
    obj = list(pyomomodel.component_data_objects(
        Objective, active=True, descend_into=True)
    if len(obj) != 1:
        raise RuntimeError("Could not identify exactly one active "
                           "Objective for model '%s' (found %d objectives)"
                           % (pyomomodel.name, len(obj)))
    return obj[0]
```

Comment (DLWoodruff) [reply mpisppy/utils/sputils.py:749]: It pays to have friends. I will make the change.

**Pull request merged**